xmtp/src/storage/encrypted_store/schema.rs

```diff
@@ -1,5 +1,75 @@
 // @generated automatically by Diesel CLI.
 
+/*
+SIMPLIFYING ASSUMPTIONS:
+- The installations in a conversation do not change after the conversation is initialized. Can sketch out plan for adding/removing installations from existing conversations later.
+- We do not include message data on invites. Ideally we start including messages directly on invites later.
+- Every state update is a DB write. This allows us to resume on cold start.
+- On cold start, we can scan the DB for UNINITIALIZED messages and resume sending them.
+- Every message and invite has a randomly generated id. This is used to deduplicate on the receiving side. If a message is received with an id that already exists in the DB, it is ignored (after advancing the ratchet state).
```

Comment:
If we are hashing the encrypted content, they should always be unique because of nonces. Even if we don't include an inner envelope that includes a timestamp like we do for V1/V2.

Reply:
> we should store individual copies of the encrypted payloads for each receiver in the DB until they are confirmed as sent,

The other option is to only persist the Session when the Network Accepts the message. That way state is rolled back appropriately on an error and can be retried later.